lib/python/bind_units.cpp

```diff
@@ -35,7 +96,28 @@ void init_units(py::module &m) {
       .def("__abs__", [](const units::Unit &self) { return abs(self); })
       .def(py::self == py::self)
       .def(py::self != py::self)
-      .def(hash(py::self));
+      .def(
+          "is_exactly_the_same",
```

Comment:
Slightly odd name. `identical`? `equals`? Maybe this should not be a method as it is mainly for internal use. Make it a free function in `scipp.units`?

Reply:
Moved it to `scipp._scipp.core.units_identical`